administration/src/bp-modules/self-service/hooks/useCardGeneratorSelfService.ts

```diff
@@ -31,12 +33,34 @@ type UseCardGeneratorSelfServiceReturn = {
   downloadPdf: (code: CreateCardsResult, fileName: string) => Promise<void>
 }
 
+const handleQueryParams = (
+  cardQueryParams: URLSearchParams
+): {
+  fullName?: string
+  birthday?: PlainDate
+  koblenzReferenceNumber?: string
+} => {
+  const fullName = cardQueryParams.get('name') ?? undefined
+  const birthday = PlainDate.safeFrom(cardQueryParams.get('geburtsdatum')) ?? undefined
+  const koblenzReferenceNumber = cardQueryParams.get('ref') ?? undefined
+  return { fullName, birthday, koblenzReferenceNumber }
+}
```

Comment:
:wrench: Please have a look at `AddCardsForm` and how card initialization is implemented there. I think the behavior there could be reused/generalized. I think it especially makes sense to reuse `initializeCardFromCSV` instead of implementing this and hardcoding the keys here again.

Reply:
yes this solution is even more flexible.
Nice find

Reply:
I assumed there is no difference, what to merge first.
I added to-do in the flutter PR.
But of course I will address this too

Reply:
Updated, but haven't tested card creation yet because of the issue https://github.com/digitalfabrik/entitlementcard/issues/1795